### PR TITLE
remove caching artifacts for datasets and fix it for vectors

### PR DESCRIPTION
**Pull request by parmeet**
This PR:
* gets rid of caching datasets list and corresponding file usage in CircleCI config since we no longer cache the datasets
* fix caching artifacts for vectors which got broken in PR #1587

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -43,7 +43,6 @@ commands:
           name: Generate CCI cache key
           command: |
             echo "$(date "+%D")" > .cachekey
-            cat .circleci/cached_datasets_list.txt >> .cachekey
       - persist_to_workspace:
           root: .
           paths:
```
[the remark this answers is not in the document]
Comment (parmeet) [reply .circleci/config.yml:48]: hmm, it seems used in caching vectors. but let me take a deeper look, we might be able to completely get rid of caching.

[the remark this answers is not in the document]
Comment (Nayef211) [reply .circleci/config.yml:48]: It looks like the caching steps have been failing for [Linux](https://app.circleci.com/pipelines/github/pytorch/text/5197/workflows/b70e051b-845e-4543-920c-a201e25ac198/jobs/174614), [MacOS](https://app.circleci.com/pipelines/github/pytorch/text/5197/workflows/b70e051b-845e-4543-920c-a201e25ac198/jobs/174592), and [Windows](https://app.circleci.com/pipelines/github/pytorch/text/5197/workflows/b70e051b-845e-4543-920c-a201e25ac198/jobs/174601) unittests.

[the remark this answers is not in the document]
Comment (parmeet) [reply .circleci/config.yml:48]: just had a look, unfortunately, it is not trivial to remove vectors caching. it is used in several places in https://github.com/pytorch/text/blob/main/test/test_build.py. I am not sure if removing the caching would significantly increase the test time?

The problem with these tests is exactly the same as with datasets before mocking i.e they are using real data to test.

[the remark this answers is not in the document]
Comment (Nayef211) [reply .circleci/config.yml]: Gotcha, in that case can we figure out how to fix this error for the time being?

```
error computing cache key: template: cacheKey:1:27: executing "cacheKey" at <checksum ".cachekey">: error calling checksum: open /root/project/.cachekey: no such file or directory
```

Could this be because we don't explicitly create the `/root/project/.cachekey` directory? I wonder if our unittests are even able to take advantage of the cache folder right now.

[the remark this answers is not in the document]
Comment (parmeet) [reply .circleci/config.yml]: yupp, I made changes to resemble to original cache before introducing the caching for datasets. 

> I wonder if our unittests are even able to take advantage of the cache folder right now.
Right, I mistakenly got rid of all the caching when removing caching for datasets in this PR https://github.com/pytorch/text/pull/1587. I guess, we might be able to remove the cache completely and just do testing without it, although I think it wouldn't hurt to have caching either. Let's land this PR to resolve the above issue, and we can follow-up later to see if we want to completely remove caching.

**Review by Nayef211 — COMMENTED** [review]
Could you rebase on the latest `main` branch and add the caching changes to `unittest_macos` as well ([code pointer](https://github.com/pytorch/text/blob/main/.circleci/config.yml.in#L454-L493))

**Comment by parmeet** [comment]
OK so I just look at some stats, and apparently having caching in place saves quite a bit of time. Below is breakdown for unit-testing in linux with python 3.7:

Time taken to execute unit-tests:
with caching: 2m 36s
without caching: 11m 50s

Time taken to save cache: 11m 23s

Time taken to restore cache: 1m 49s

Total running time for job with cache present: 9m 5s
Total running time for job without cache present (this also includes time to save it): 27m 42s

Since the cache is saved/generated on weekly basis, we can ignore the time taken for saving it (one time cost).

So overall we reduce the job time by ~43% by enabling cache (from ~16m without cache to ~9m with cache)

**Comment by parmeet** [comment]
> Could you rebase on the latest `main` branch and add the caching changes to `unittest_macos` as well ([code pointer](https://github.com/pytorch/text/blob/main/.circleci/config.yml.in#L454-L493))

done!

**Comment by Nayef211** [comment]
> OK so I just look at some stats, and apparently having caching in place saves quite a bit of time. Below is breakdown for unit-testing in linux with python 3.7:
> 
> Time taken to execute unit-tests: with caching: 2m 36s without caching: 11m 50s
> 
> Time taken to save cache: 11m 23s
> 
> Time taken to restore cache: 1m 49s
> 
> Total running time for job with cache present: 9m 5s Total running time for job without cache present (this also includes time to save it): 27m 42s
> 
> Since the cache is saved/generated on weekly basis, we can ignore the time taken for saving it (one time cost).
> 
> So overall we reduce the job time by ~43% by enabling cache (from ~16m without cache to ~9m with cache)

That's a great speedup @parmeet! Looks like it's worthwhile keeping our cache moving forward then. Just want to double check that the main thing we are saving in this cache are the pre-trained vector weights right? Or are there other artifacts we're also caching?

**Comment by parmeet** [comment]
> Just want to double check that the main thing we are saving in this cache are the pre-trained vector weights right? Or are there other artifacts we're also caching?

Yupp, that's right.
